AIPscan/API/namespace_data.py

```diff
@@ -114,3 +114,43 @@ def get(self, storage_service_id, file_type=None, limit=20):
             storage_service_id=storage_service_id, file_type=file_type, limit=limit
         )
         return file_data
+
+
+@api.route("/storage_service/<storage_service_id>/file-format")
+class AIPsByFormatList(Resource):
+    @api.doc(
+        "list_aips_by_format",
+        params={
+            "file_format": {
+                "description": "File format name (must be exact match)",
+                "in": "query",
+                "type": "str",
+            }
+        },
+    )
+    def get(self, storage_service_id):
+        """AIPs containing file format"""
+        file_format = request.args.get("file_format")
+        if not file_format:
+            return {"success": "False", "error": "Must specify a file format."}
```

Comment:
I will change this to returning an empty list for no match, as I agree that that's much cleaner. As noted above, I had been thinking of the API as "nice to have" rather than a primary goal, hence this rather simple but crude error response.

Reply:
The design intent here was to signal to the caller that their call was invalid because of a missing required parameter. I completely agree that if the parameter is included but no results are found, the best response is to return an empty list. That's what's returned currently if you pass a value for which no results are found, e.g.:

```
{
  "StorageName": "Docker AMSS",
  "Format": "some invalid value",
  "AIPs": []
}
```

Your question about headers is interesting, and a good one that I think warrants some investigation. My `return {"success": "False", "error": "Must specify a file format."}` solution was a bit crude and the concerns you raise around handling outputs are valid ones. I'll plan to do a bit of reading on FlaskX to see if there's a better way to handle missing params.

Reply:
Thanks Tessa - yeah, I think the intention is clear, unless I suggested otherwise? I think it's a nice idea too - and really emphasizes your desire for usability. 

I can't seem to find the comment, it might have arrived by email first and then you re-drafted? But I believe you mentioned viewing the API as a sort of nice to have - that's a pretty fair comment at this stage and I might have better remembered that in terms of guidance during review too. If it helps I might leverage that perspective for now and say, let's leave the API fairly minimal in what it's doing and returning and we can then take a more opinionated approach to shaping it a little further down the line. We're not very many iterations down the line yet, and we should each definitely put our heads together about what we want to do as well as avoid here.

Reply:
Oh a second recording, you're totally right and I missed the gist of what you're getting at.

I did do some re-drafting of comments and might have lost that API bit, so I'm glad it came through regardless. Sounds like a good plan to me.